### PR TITLE
HW 35 Polina

### PR DESCRIPTION
**Pull request by polishukova**

### CODE REVIEW

**Comment by polishukova** [comment]
I don't know, why is older commits here..

File: hw-35.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    return namesStr
+}
+
+console.log(getUserNames(users))
+
+//     2. Посчитать общее количнство машин у пользователей.
+
+const sumCars = (users: User[]) => {
+    const carsArr = users.map(user => user.cars)
+    return carsArr.length
```
Comment (jathpr) [inline hw-35.ts:53]: wrong

File: hw-35.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    return usersArr
+}
+
+console.log(isEducationed(users));
+
+
+//     4. Создать функцию, которая бы принимала массив пользователей и отфильтровывала пользователей на наличие животных.
+
+const isHasAnimals= (users: User[]) => {
+    const usersArr = users.filter(user => user.animals)
```
Comment (jathpr) [inline hw-35.ts:71]: potentially you can check empty array case too

File: hw-35.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+console.log(isHasAnimals(users));
+
+//     5. Создать функцию, которая бы принимала массив пользователей и отдавала бы  строку с названиями марок автомобилей через запятую
+
+const getCarNames = (users: User[]) => {
+    const isCarArr = users.filter(user => user.cars)
+    const carArr = isCarArr.map(user => user.cars)
+    const carStr = carArr.join (',')
```
Comment (jathpr) [inline hw-35.ts:82]: looks like it would be an array of arrays, not an array of strings
you can try to use Set here possible with reduce

**Review by jathpr — CHANGES_REQUESTED** [review]
time to fix git has come)
hw looks good for me

**Pull request closed without merging**